src/remote/server/server.cpp

```diff
@@ -709,11 +709,30 @@ class ServerAuth : public GlobalStorage, public ServerAuthBase
 					}
 				}
 
-				if (send->p_acpt.p_acpt_type & pflag_compress)
-					authPort->initCompression();
-				authPort->send(send);
-				if (send->p_acpt.p_acpt_type & pflag_compress)
+				if ((send->p_acpt.p_acpt_type & pflag_compress) == 0)
+				{
+					authPort->send(send); //throw?
+				}
+				else
+				{
+					// Configuration of port for compression usage (1-3).
+					fb_assert(send->p_acpt.p_acpt_type & pflag_compress);
+
+					// 1. Prepare compression. Enable inflate.
+					authPort->initCompression(); //throw
+
+					// 2. Send uncompressed answer.
+					authPort->send(send); //throw?
+
+					// ATTENTION:
+					//  In this moment server may already got the compressed
+					//  packet from client and is waiting for finish of this
+					//  not completed answer to previous packet!
+
+					// 3. Enabled deflate.
 					authPort->port_flags |= PORT_compressed;
+				}//else
+
```

Comment:
You remove duplication of condition check but add one for function call. Very doubtfully...
And code becomes 5 times longer than before. That's also inconvenient - we still do not have video walls from Bradbury's 451 as monitors :) Personally I dislike this change.